spring/boot2-autoconfigure/src/main/java/com/linecorp/armeria/spring/ArmeriaAutoConfiguration.java

```diff
@@ -16,132 +16,35 @@
-            return null;
-        }
-
-        final ServerBuilder serverBuilder = Server.builder();
-
-        final List<Port> ports = armeriaSettings.getPorts();
-        if (ports.isEmpty()) {
-            serverBuilder.port(new ServerPort(DEFAULT_PORT.getPort(), DEFAULT_PORT.getProtocols()));
-        } else {
-            configurePorts(serverBuilder, ports);
-        }
-
-        final DocServiceBuilder docServiceBuilder = DocService.builder();
-        docServiceConfigurators.ifPresent(
-                configurators -> configurators.forEach(
-                        configurator -> configurator.configure(docServiceBuilder)));
-
-        final String docsPath = armeriaSettings.getDocsPath();
-        configureServerWithArmeriaSettings(serverBuilder, armeriaSettings,
-                                           meterRegistry.orElse(Metrics.globalRegistry),
-                                           healthCheckers.orElseGet(Collections::emptyList),
-                                           meterIdPrefixFunction.orElse(
-                                                   MeterIdPrefixFunction.ofDefault("armeria.server")));
+    static class NonReactiveWebApplicationCondition extends NoneNestedConditions {
 
-        armeriaServerConfigurators.ifPresent(
-                configurators -> configurators.forEach(
-                        configurator -> configurator.configure(serverBuilder)));
-
-        armeriaServerBuilderConsumers.ifPresent(
-                consumers -> consumers.forEach(
-                        consumer -> consumer.accept(serverBuilder)));
-
-        if (!Strings.isNullOrEmpty(docsPath)) {
-            serverBuilder.serviceUnder(docsPath, docServiceBuilder.build());
+        NonReactiveWebApplicationCondition() {
+            super(ConfigurationPhase.PARSE_CONFIGURATION);
         }
 
-        final Server server = serverBuilder.build();
-
-        server.start().handle((result, t) -> {
-            if (t != null) {
-                throw new IllegalStateException("Armeria server failed to start", t);
-            }
-            return result;
-        }).join();
-        logger.info("Armeria server started at ports: {}", server.activePorts());
-        return server;
-    }
-
-    /**
-     * A user can configure a {@link Server} by providing an {@link ArmeriaServerConfigurator} bean.
-     */
-    @Bean
-    @ConditionalOnProperty("server.shutdown")
-    public ArmeriaServerConfigurator gracefulShutdownServerConfigurator(
-            @Value("${server.shutdown}") String shutdown,
-            @Value("${spring.lifecycle.timeout-per-shutdown-phase:30s}") Duration duration) {
-        if (GRACEFUL_SHUTDOWN.equalsIgnoreCase(shutdown)) {
-            return sb -> sb.gracefulShutdownTimeout(duration, duration);
-        } else {
-            return sb -> {};
-        }
+        @ConditionalOnWebApplication(type = ConditionalOnWebApplication.Type.REACTIVE)
+        static class ReactiveWebApplication {}
```

Comment:
Question: Could you tell me how this works please? 😅

Reply:
Looking at `@ConditionalOnWebApplication` alone, when the application is launched, it determines whether to create a bean according to the web application type. This means that if users has `webflux` as a dependency and there is no explicit configuration, the web application type is reactive.

In this case, it is a negative condition because I used `NoneNestedConditions`. So when the web application type is reactive, Spring don't make `ArmeriaAutoConfiguration` a bean.

Reply:
These settings can also be seen in [spring boot code](https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/client/ RestTemplateAutoConfiguration.java#L77).

Reply:
Thanks for the education. 😄